lib/utils.nix

```diff
@@ -172,6 +172,41 @@ rec {
     in
     lib.literalExpression exp;
 
+  /**
+    Convert one or several conditional warnings to a final warning list.
+
+    # Example
+
+    ```nix
+    warnings = mkWarnings [
+      {
+        condition = plugins.foo.settings.barIntegration && (!plugins.bar.enable);
+        message = "`barIntegration` is enabled but the `bar` plugin is not."
+      }
+    ];
+    ```
+
+    # Type
+    ```
+    mkWarnings :: String -> List -> List
+    ```
+  */
+  mkWarnings =
+    scope: w:
+    let
+      warnings = if !lib.isList w then [ w ] else w;
+
+      processWarning =
+        warning:
+        lib.optional (warning.condition or true) (
+          let
+            message = if lib.isString warning then warning else warning.message;
```

Comment:
```suggestion
            message = warning.message or warning;
```
?

Reply:
Also: this would be a great place to strip trailing whitespace, e.g. using [`lib.trim`](https://noogle.dev/f/lib/strings/trim)